src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,918 @@
+    138, // REGION_ROMANI_RANCH
+    171, // REGION_DEKU_PALACE
+    146, // REGION_WOODFALL
+    119, // REGION_CLOCK_TOWN
+    77,  // REGION_SNOWHEAD
+    106, // REGION_IKANA_GRAVEYARD
+    120, // REGION_IKANA_CANYON
+    73,  // REGION_GORON_VILLAGE
+    99,  // REGION_STONE_TOWER
 };
-
-s16 D_8082B5CC[] = {
-    0x0005, 0x0004, 0x0006, 0x0000, 0x0008, 0x0005, 0x0004, 0x0006, 0x0000, 0x0008,
+s16 sGreatFairySpawnRegions[] = {
+    REGION_CLOCK_TOWN, REGION_WOODFALL, REGION_SNOWHEAD, REGION_GREAT_BAY, REGION_IKANA_CANYON,
+    REGION_CLOCK_TOWN, REGION_WOODFALL, REGION_SNOWHEAD, REGION_GREAT_BAY, REGION_IKANA_CANYON,
 };
 
-u16 D_8082B5E0[] = {
-    0xAF, 0xB3, 0xAA, 0xB1, 0xA9, 0xB2, 0xA8, 0xB0, 0xAC, 0xAE,
+void KaleidoScope_DrawWorldMap(PlayState* play) {
+    s16 sceneId;
+    s16 t;
+    s16 n;
+    s16 j;
+    s16 k;
+    s16 i;
+    PauseContext* pauseCtx = &play->pauseCtx;
+    s16 rectLeft;
+    s16 rectRight;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    KaleidoScope_SetCursorVtx(pauseCtx, pauseCtx->cursorSlot[PAUSE_MAP] * 4, pauseCtx->mapPageVtx);
+
+    if ((pauseCtx->pageIndex == PAUSE_MAP) && (pauseCtx->state == 6) &&
+        ((pauseCtx->unk_200 == 0) || (pauseCtx->unk_200 == 3)) && YREG(6) && (pauseCtx->state != 7) &&
+        !((pauseCtx->state >= 8) && (pauseCtx->state <= 0x12))) {
+
+        func_8012C628(play->state.gfxCtx);
+
+        gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_POINT);
+        gDPLoadTLUT_pal256(POLY_OPA_DISP++, gWorldMapImageTLUT);
+        gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_RGBA16);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+
+        // Loop over yPos (t), textureIndex (j)
+        for (t = 62, j = 0; j < 16; j++, t += 8) {
+            gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + j * (216 * 8), G_IM_FMT_CI, G_IM_SIZ_8b, 216,
+                                8, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                G_TX_NOLOD, G_TX_NOLOD);
+
+            rectLeft = 204;
+            rectRight = rectLeft + (216 << 2);
+            gSPTextureRectangle(POLY_OPA_DISP++, rectLeft, t << 2, rectRight, (t << 2) + (8 << 2), G_TX_RENDERTILE, 0,
+                                0, 1 << 10, 1 << 10);
+        }
+
+        func_8012C8AC(play->state.gfxCtx);
+
+    } else {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_POINT);
+        gDPLoadTLUT_pal256(POLY_OPA_DISP++, gWorldMapImageTLUT);
+        gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_RGBA16);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+        gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[204], 32, 0);
+
+        // Loop over textureIndex (i, k), vtxIndex (j)
+        for (i = 0, k = 0, j = 0; i < 8; i++, k++, j += 4) {
+            gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + k * (216 * 9), G_IM_FMT_CI, G_IM_SIZ_8b, 216,
```

Comment:
Would mention this is done in chunks, probably worth making some defines (may be relatable to TMEM_SIZE?)